Cargo.lock

```diff
@@ -1,5 +1,20 @@
 # This file is automatically @generated by Cargo.
```

Comment:
Quite a few dependencies being added here, we should audit which of those are actually needed :)

Reply:
I updated some dependencies so they versions match which I can push but not sure how to do a proper audit

Reply:
This tool is always a good start: https://github.com/est31/cargo-udeps

Reply:
Can be done in follow-up PR please :)